docs/user_manual.md

```diff
@@ -1285,12 +1285,16 @@ rule (X +Int Y) +Int Z => X +Int (Y +Int Z) [simplification]
 A simplification rule is only applied when the current side condition _implies_
 the `requires` clause of the rule, like function definition rules.
 
-**Order**: Simplification rules are applied after definition rules, if no
-definition rule did apply. The `simplification` attribute accepts an optional
-integer argument which is the rule's _priority_; if the optional argument is not
-specified, it is equivalent to a priority of 50. Simplification rules are
-applied in order of their priority. `simplification` rules may not have the
-`priority` attribute.
+**Order**: The `simplification` attribute accepts an optional integer argument
+which is the rule's _simplification priority_; if the optional argument is not
+specified, it is equivalent to a simplification priority of 50. Backends
+_should_ attempt simplification rules in order of their _simlpification
+priority_, but are not required to do so; in fact, the backend is free to apply
+`simplification` rules at _any time_. Because of this, users must ensure that
+simplification rules are sound regardless of their order of application. This
+differs from the `priority` attribute in that rules with the `priority`
+attribute _must_ be applied in their priority order by the backend. It is an
+error to have the `priority` attribute on a `simplification` rule.
```

Comment:
You say that simplification rules should be sound regardless of the order they apply.
Then why do I have the option to add a priority? What changes? Do I get a more deterministic execution when possible?
It's not clear from this explanation why should I bother adding those priorities.

@ana-pantilie maybe you have some suggestions here?

Reply:
Because you want the backend to try one simplification path _before_ it tries another one. It's a heuristic you can supply to the backend.